scipy/interpolate/_interpolate.py

```diff
@@ -689,51 +693,52 @@ def extend(self, c, x):
         if c.size == 0:
             return
 
-        dx = np.diff(x)
-        if not (np.all(dx >= 0) or np.all(dx <= 0)):
-            raise ValueError("`x` is not sorted.")
+        with self._c_lock:
+            dx = np.diff(x)
```

Comment:
Why does manipulating `x` need a lock? In general, I've to admit I'm confused on what needs locking and what does not.

Reply:
This is locking the whole region spanned from lines 697 to 741, which has several references to `self.c`